caddyhttp/fastcgi/fastcgi.go

```diff
@@ -110,8 +149,12 @@ func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error)
 
 			// Log any stderr output from upstream
 			if fcgiBackend.stderr.Len() != 0 {
+				persistentConnections.Lock()
 				// Remove trailing newline, error logger already does this.
 				err = LogError(strings.TrimSuffix(fcgiBackend.stderr.String(), "\n"))
```

Comment:
Ah you are right. I wanted to include the close (otherwise another thread might try to reuse the closed connection) but not the log. Even better to first lock and delete, than close. Thx.